homeassistant/components/control4/light.py

```diff
@@ -73,40 +73,71 @@ async def async_update_data_dimmer():
+
+    entity_list = []
     for item in items_of_category:
-        if item["type"] == CONTROL4_ENTITY_TYPE:
-            item_name = item["name"]
-            item_id = item["id"]
-            item_parent_id = item["parentId"]
-            item_is_dimmer = item["capabilities"]["dimmer"]
-
-            if item_is_dimmer:
-                item_coordinator = dimmer_coordinator
+        try:
+            if item["type"] == CONTROL4_ENTITY_TYPE:
+                item_name = item["name"]
+                item_id = item["id"]
+                item_parent_id = item["parentId"]
+
+                item_manufacturer = None
+                item_device_name = None
+                item_model = None
+
+                for parent_item in items_of_category:
+                    if parent_item["id"] == item_parent_id:
+                        item_manufacturer = parent_item["manufacturer"]
+                        item_device_name = parent_item["name"]
+                        item_model = parent_item["model"]
             else:
-                item_coordinator = non_dimmer_coordinator
-
-            for parent_item in items_of_category:
-                if parent_item["id"] == item_parent_id:
-                    item_manufacturer = parent_item["manufacturer"]
-                    item_device_name = parent_item["name"]
-                    item_model = parent_item["model"]
-            async_add_entities(
-                [
-                    Control4Light(
-                        entry_data,
-                        entry,
-                        item_coordinator,
-                        item_name,
-                        item_id,
-                        item_device_name,
-                        item_manufacturer,
-                        item_model,
-                        item_parent_id,
-                        item_is_dimmer,
-                    )
-                ],
-                True,
+                continue
+        except KeyError as exception:
+            _LOGGER.error(
+                "Unknown device properties received from Control4: %s %s",
+                exception,
+                item,
+            )
+            continue
+
+        if item_id in dimmer_coordinator.data:
+            item_is_dimmer = True
+            item_coordinator = dimmer_coordinator
+        elif item_id in non_dimmer_coordinator.data:
+            item_is_dimmer = False
+            item_coordinator = non_dimmer_coordinator
+        else:
+            director = entry_data[CONF_DIRECTOR]
+            item_variables = await director.getItemVariables(item_id)
+            _LOGGER.warning(
+                "Couldn't get light state data for %s, skipping setup", item_name,
+            )
+            _LOGGER.debug(
+                "Couldn't get light state data for %s, skipping setup. Available variables from Control4: %s",
+                item_name,
+                item_variables,
             )
```

Comment:
We should only log once here for the same error. If something is broken, `warning` is fine so we can drop the second `debug`.  If nothing is actually broken than go with `debug`
```suggestion
            _LOGGER.warning(
                "Couldn't get light state data for %s, skipping setup. Available variables from Control4: %s",
                item_name,
                item_variables,
            )
```